### PR TITLE
refactor: world v2 entity record

### PR DESCRIPTION
**Pull request by 0xxlegolas**
**Description**

- simplified implementation for world-v2 
- entity record functions

### CODE REVIEW

File: mud-contracts/world-v2/src/systems/constants.sol [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//SPDX-License-Identifier: MIT
```
Comment (tinkrtailor) [inline mud-contracts/world-v2/src/systems/constants.sol:1]: Should this file be at the top level of the source code directory?

File: mud-contracts/world-v2/src/systems/entity-record/EntityRecordSystem.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   * @dev sets the name of an entity
+   * @param entityId the id of a in game entity referred as smart object id
+   * @param name the name of the entity
+   */
+  function setName(uint256 entityId, string memory name) public {
+    EntityRecordMetadata.setName(entityId, name);
+  }
+
+  /**
+   * @dev sets the description of an entity
```
Comment (tinkrtailor) [inline mud-contracts/world-v2/src/systems/entity-record/EntityRecordSystem.sol]: Is this doc comment correct?

File: mud-contracts/world-v2/mud.config.ts [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
+        recordExists: "bool",
       },
-      key: [],
+      key: ["entityId"],
+    },
+    EntityRecordMetadata: {
+      schema: {
+        entityId: "uint256",
+        name: "string",
+        dappURL: "string",
```
Comment (tinkrtailor) [inline mud-contracts/world-v2/mud.config.ts:28]: Why does every entity have a dappurl in its metadata?

Comment (0xxlegolas) [reply mud-contracts/world-v2/mud.config.ts:28]: Each entity can have an associated external DApp. For example, any entity that can be smart-deployed or act as a smart turret, when customized by the builder, requires a URL to access the custom-built DApp.

**Review by tinkrtailor — APPROVED** [review]

**Pull request merged**